master/internal/api_experiment.go

```diff
@@ -1922,6 +1923,312 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 	return nil
 }
 
+func scanMetricName(filter string, index int) string {
+	// Determine the name of the metric from the filter string
+
+	metricName := ""
+	for index < len(filter) {
+		c := string(filter[index])
+
+		// If we have reached an operator or space then we have parsed through the entire metric name
+		if c == "<" || c == ":" || c == " " || c == "~" || c == "=" || c == ">" {
+			break
+		}
+		metricName += c
+		index++
+	}
+	return metricName
+}
+
+func buildQuery(filter string) (string, error) {
+	// Builds the sql query for a filter string
+
+	// The prefix for column names mapping to
+	// experiment table values
+	experimentColumnPrefix := "experiment."
+
+	// Prefixes for column names matching to metric related tables
+	metricPrefixes := map[string]string{
+		"validation.": "(e.validation_metrics->>'%s')::float8",
```

Comment:
An assumption is made here that all hyperparameter and validation metric values will be numeric. If that is not the case then this will need to be updated to determine the correct value type.